Miriam1986 patch 1 (#33)
Laura-Novich-OBW/student-showcase: student-work/miriam-rosenberg/api-final-project/POST-order-meal.md

# POST/lunch

> The POST request is created when a confirmed burger meal order comes in. Current `"mealType"` is limited to lunch only for POC purposes and will be developed to include additional menu items. The burger meal JSON object is included in the below code snippet. 

Reviewer: ```suggestion
The POST request is sent when an order is submitted. Current `"mealType"` is limited to lunch only for POC purposes and will be developed to include additional menu items. The burger meal JSON object is included in the below code snippet. 
```

```
curl -H "Content-Type: application/json" -X POST -d'{

"tableNumber":99,
"mealType":"lunch",
   "mealCat":{
  	"main":"burgerMeal",
  	"burger":{
     	"pattyType":"beef",
     	"pattyQty":1,
     	"pattyWeightG":100,
     	"pattyCook":"MR",
     	"bunType":"wholeWheat",
     	"condiment1":"ketchup",
     	"condiment2":"secretSauce",
     	"topping1":"lettuce",
     	"topping2":"pickles",
     	"topping3":"None",
     	"topping4":"None",
  	},
  	"sides":{
     	"side1":{
        	"type":"frenchFries",
        	"size":"large"
     	},
     	"side2":{
        	"type":"none",
        	"size":""
     	}
  	},
  	"drink":{
     	"type":"Coke",
     	"size":"large",
     	"ice":true
```

## Code snippet explanation

> The lunch burger meal consists of burger, bun, condiment, topping, side, and drink.

Reviewer: ```suggestion
The lunch burger meal consists of a burger, with a bun, condiments, and toppings, sides, and a drink.
```

| mealType      	|  string    	|  "lunch"             	|  Y   	| Meal type depending on the hour. | "breakfast", "lunch", "dinner"|
| mealCat      	|  string    	|    "burgerMeal" |    Y   | Type of a meal ordered by a customer.| Future variables are set menu items.|

### Burger

`"burger"` is an object which is a part of the meal consisting of properties listed below. All properties are mandatory. All `"none"` values are valid wherever indicated in the Value Variables column. 

| Property Name 	| Data Type 	| Values                 |Mandatory | Definition                           	|Value Variables|
|-----------------|-------------|------------------------|----------|--------------------------------------	|----------------|
| pattyType       | string      | "beef"	               |   Y      | Type of patty. | "beef", "lamb", "chickenBreast", "fish", and "bean"|
| pattyQty        | int         | 1	                     |   Y      |Number of patties in the burger. Additional patties are at an extra cost. | 1, 2, and 3 |
| pattyWeightG    | int         | 100	                   |   Y      |Patty weight in grams. The price increases with weight.|100, 170, and 250|

> | pattyCook       | string      | "MR"                   |   Y      |The way a patty is cooked.| "MR" medium-rare, "M" medium, "MW" medium-well, "WD" well-done. **Note:** chicken breast, fish and bean patty can only have "WD" well-done option.|

Reviewer: ```suggestion
| pattyCook       | string      | "MR"                   |   Y      |The way a patty is cooked.| "MR" medium-rare, "M" medium, "MW" medium-well, "WD" well-done. **Note:** chicken breast, fish and bean patty only have "WD" well-done option.|
```
